include/mitsuba/render/mesh.h

```diff
@@ -303,19 +306,18 @@ class MTS_EXPORT_RENDER Mesh : public Shape<Float, Spectrum> {
                 // Evaluate spectral upsampling model from stored coefficients
                 UnpolarizedSpectrum c0, c1, c2;
 
-                c0 = srgb_model_eval<UnpolarizedSpectrum>(v0, si.wavelengths);
-                c1 = srgb_model_eval<UnpolarizedSpectrum>(v1, si.wavelengths);
-                c2 = srgb_model_eval<UnpolarizedSpectrum>(v2, si.wavelengths);
+                c0 = srgb_model_eval<UnpolarizedSpectrum>(srgb_model_fetch(v0), si.wavelengths);
```

Comment:
What if the 3D data isn't representing color? I think this should only happen when calling `eval()` and not `eval_3()`. Could we differentiate those two cases?